### PR TITLE
Refactor header toolbar items to use @wordpress/data hooks

### PR DESCRIPTION
**Pull request by diegohaz**
Refactor suggested by @gziolo on https://github.com/WordPress/gutenberg/pull/22354#discussion_r435001372

## How to test?

Check if Undo/Redo and the Table Of Contents buttons on the header toolbar are working properly.

### CODE REVIEW

**Review by gziolo — APPROVED** [review]
Nice, thanks 😊

File: packages/editor/src/components/editor-history/redo.js [hunk 1]
```diff
@@ -3,17 +3,20 @@
...
+import { useSelect, useDispatch } from '@wordpress/data';
 import { displayShortcut } from '@wordpress/keycodes';
 import { redo as redoIcon } from '@wordpress/icons';
 import { forwardRef } from '@wordpress/element';
 
-function EditorHistoryRedo( { hasRedo, redo, innerRef, ...props } ) {
+function EditorHistoryRedo( props, ref ) {
+	const hasRedo = useSelect( ( select ) =>
+		select( 'core/editor' ).hasEditorRedo()
+	);
```
Comment (ZebulanStanphill) [inline packages/editor/src/components/editor-history/redo.js:14]: Noting that you forgot to add a dependency array to this and the other `useSelect` calls to memoize the callbacks.

Comment (diegohaz) [reply packages/editor/src/components/editor-history/redo.js:14]: Oh, thanks for noticing! I'll create a follow-up PR.

Comment (gziolo) [reply packages/editor/src/components/editor-history/redo.js:14]: We really need a lint rule, it’s impossible to catch all cases in review.

**Pull request merged**